src/fastfield/serializer/gcd.rs

```diff
@@ -0,0 +1,226 @@
+use fastfield_codecs::{FastFieldDataAccess, FastFieldStats};
+use gcd::Gcd;
+use libdivide::Divider;
+
+pub const GCD_DEFAULT: u64 = 1;
+
+fn compute_gcd(vals: &[u64], base: u64) -> u64 {
+    let mut gcd = (vals[0] - base).gcd(vals[1] - base);
+
+    for el in vals.iter().map(|el| el - base) {
+        gcd = gcd.gcd(el);
```

Comment:
if gcd == 1 {
    return 1;
}

Reply:
Is it something we might want to make faster?

In addition to the short cut above, another easy win is:
```
if el % gcd == 0 { //< that one should rely on libdivide.
    continue;
}
```